devtools/pyroscope_pudl_etl.sh

```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/bash
+
+pyroscope exec --log-level info --spy-name pyspy \
```

Comment:
I've been using [py-spy](https://github.com/benfred/py-spy) for profiling because you can save the profile data as a json. I don't think pyroscope allows you to save the data (I could be wrong).

Reply:
Do you have a script that will profile the whole ETL process in one go using py-spy that we could substitute in here? I had trouble getting pyroscope up and running reliably.

Reply:
I do! I can add it to this PR if you like.

Reply:
Yeah, if you've got a script and would like to add it to this PR and replace the `pyroscope_pudl_etl.sh` script that would be great. We can figure out how to integrate it into the CI once we've got it running in cloudland.